src/controllers/controllerengine.cpp

```diff
@@ -330,6 +341,25 @@ bool ControllerEngine::internalExecute(QScriptValue thisObject,
                      scriptCode);
 
         scriptErrorDialog(error);
+        m_pEngine->clearExceptions();
+        return false;
+    }
+    return true;
+}
+
+/* -------- ------------------------------------------------------
+Purpose: Evaluate & run script code
+Input:   'this' object if applicable, Code string
+Output:  false if an exception
+-------- ------------------------------------------------------ */
+bool ControllerEngine::internalExecute(QScriptValue thisObject,
+                                       const QString& scriptCode) {
+    // A special version of safeExecute since we're evaluating strings, not actual functions
+    //  (execute() would print an error that it's not a function every time a timer fires.)
+    if (m_pEngine == nullptr)
+        return false;
+
+    if (!syntaxIsValid(scriptCode)) {
```

Comment:
This call along with the wrapper generation should be done earlier, just after evaluating the QScriptValue as string places like: 
ControllerEngine::beginTimer